instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/TracingBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public abstract class TracingBuilder<
```

Comment:
This is probably a good opportunity to revisit https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3156 - should it really be `TracingBuilder`? Maybe `TelemetryBuilder`? (and `FooTelemetryBuilder`, `FooTelemetry`)

Reply:
Yeah kept this consistent with what we have for now but realized we should rename to something like FooInstrumentation